lib/api/CDataSummarizationJsonSerializer.cc

```diff
@@ -117,19 +118,19 @@ void CDataSummarizationJsonSerializer::addToJsonStream(TGenericLineWriter& write
     writer.StartObject();
 
     writer.Key(JSON_NUM_COLUMNS_TAG);
-    writer.Uint64(m_Frame.numberColumns());
+    writer.Uint64(m_NumberColumns);
 
     writer.Key(JSON_COLUMN_NAMES_TAG);
     writer.StartArray();
-    for (const auto& columnName : m_Frame.columnNames()) {
-        writer.String(columnName);
+    for (std::size_t i = 0; i < m_NumberColumns; ++i) {
+        writer.String(m_Frame.columnNames()[i]);
```

Comment:
Why can't range for loops be used here any more?

Reply:
This was an oversight in the original implementation. The data frame contains extra columns which it uses to cache information during training. These should not be serialised. So here `m_NumberColumns < m_Frame.columnNames().size()`. On reflection, it is worth adding a comment for this.

Reply:
See [2edd458](https://github.com/elastic/ml-cpp/pull/1879/commits/2edd458f6acd57dd4da9073af66de812d78afa5d).